packages/discovery-react-components/src/components/SearchFacets/SearchFacets.tsx

```diff
@@ -209,15 +209,18 @@ const SearchFacets: FC<SearchFacetsProps> = ({
     if (onChange) {
       onChange(event);
     }
+
     setFacetSelectionState({ filterFields: [], filterDynamic: [] });
     setCollectionSelectionState([]);
-    // We should update to not select with a click
+
+    // TODO We should update to not select with a click
     // when Carbon MultiSelect selection can be controlled and Downshift's action props are exposed
     (
       document.querySelectorAll(
-        `.${settings.prefix}--list-box__selection--multi`
+        `#${facetsId} .${settings.prefix}--tag--filter [role="button"]`
```

Comment:
It's not great that we still can't programmatically call MultiSelects internal `clearSelection()`, so we're left with simulated mouse click. However, previous code was _not_ sandboxed to the current component and affects all such controls on the entire page. The changes here use `facetsId` so this only affects children of the current component.

Also had to change the button selector, due to Carbon changes.